ee/clickhouse/queries/cohort_query.py

```diff
@@ -0,0 +1,261 @@
+                        (_event, _period)
+                    ] = f"event_{idx_label + 1}_from_{self.get_date_column(_start_date)}_to_{self.get_date_column(_end_date)}"
+
+            elif prop.type == "performed_event_regularly":
+                # handle regularly intervals
+                pass
+
+        return _events
+
+    def events_in_period_to_query(self) -> List[Tuple[Event, Period]]:
+        return list(self.events_in_period_to_query_column_names.keys())
+
+    def get_event_in_period_column(self, event: Event) -> ColumnName:
+        return self.events_in_period_to_query_column_names[event]
+
+    @cached_property
+    def first_time_activity_events_to_query_column_names(self) -> Dict[Tuple[Event, Period], str]:
+        _first_time_activity = {}
+        for idx, prop in enumerate(self.filter.property_groups.flat):
+            idx_label = idx
+            if (
+                prop.type == "performed_event_first_time"
+                and prop.key is not None
+                and prop.event_type is not None
+                and prop.time_interval is not None
+                and prop.time_value is not None
+            ):
+                _first_time_activity[
+                    ((prop.key, prop.event_type), (prop.time_value, prop.time_interval))
+                ] = f"first_time_activity_{idx_label}"
+
+        return _first_time_activity
+
+    def get_first_time_activity_event_in_period_column(self, event: Tuple[Event, Period]) -> ColumnName:
+        return self.first_time_activity_events_to_query_column_names[event]
+
+    @cached_property
+    def sequences_to_query_column_names(self) -> Dict[Tuple[Event, Event, Period], str]:
+        _sequences = {}
+        for idx, prop in enumerate(self.filter.property_groups.flat):
+            idx_label = idx
+            if (
+                prop.type == "performed_event_sequence"
+                and prop.key is not None
+                and prop.event_type is not None
+                and prop.seq_event is not None
+                and prop.seq_event_type is not None
+                and prop.seq_time_interval is not None
+                and prop.seq_time_value is not None
+            ):
+                _sequences[
+                    (
+                        (prop.key, prop.event_type),
+                        (prop.seq_event, prop.seq_event_type),
+                        (prop.seq_time_value, prop.seq_time_interval),
+                    )
+                ] = f"sequence_{idx_label}"
+
+        return _sequences
+
+    def get_sequence_column(self, sequence: Tuple[Event, Event, Period]) -> ColumnName:
+        return self.sequences_to_query_column_names[sequence]
+
+    def person_columns_to_query(self) -> Set[ColumnName]:
+        return set(
+            property_name for property_name, _, _, _ in self._used_properties_with_type_and_operator("person", "OR")
+        )
+
+    @cached_property
+    def is_using_person_properties_in_or(self) -> bool:
```

Comment:
When person-on-events is complete, we'll only need to join person table when a person filter is in an OR clause, otherwise we can use the person properties directly on events table to filter

Reply:
I'd worry some about using the person on events to filter the persons. I think because of the time-based nature of the person-on-events info, we still want to use the actual person object when filtering persons.

Reply:
Rick, that's the breaking change we are okay with making when person-on-events comes, to get all those perf advances

Reply:
I agree with you when we're talking about filtering events on persons properties, but I think there's a nuance here when we're filtering persons. If you filter persons based on the person-on-events properties, then the user is going to be seeing inconsistent results based on the specific events that happen to be used in the query.

Reply:
Ah, I see what you mean!

The latter I think is what users might actually want: "Filter on people who had paid_status=unpaid at the time this event occurred".

Agree that there's some ambiguity here, and _maybe_ joining might not be the thing we want, in the regular OR case as well.. But we can cross this when we get to person-on-events.